obs-studio-client/source/sceneitem.cpp

```diff
@@ -265,6 +267,136 @@ Nan::NAN_METHOD_RETURN_TYPE osn::SceneItem::SetSelected(Nan::NAN_METHOD_ARGS_TYP
 	sid->isSelected      = selected;
 }
 
+Nan::NAN_METHOD_RETURN_TYPE osn::SceneItem::IsStreamVisible(Nan::NAN_METHOD_ARGS_TYPE info)
+{
+	osn::SceneItem* item = nullptr;
+	if (!Retrieve(info.This(), item)) {
+		return;
+	}
+
+	SceneItemData* sid = CacheManager<SceneItemData*>::getInstance().Retrieve(item->itemId);
+
+	if (sid && !sid->streamVisibleChanged) {
+		info.GetReturnValue().Set(utilv8::ToValue(item->IsStreamVisible));
+		return;
+	}
+
+	auto conn = GetConnection();
+	if (!conn)
+		return;
+
+	std::vector<ipc::value> response =
+	    conn->call_synchronous_helper("SceneItem", "IsStreamVisible", std::vector<ipc::value>{ipc::value(item->itemId)});
+
+	if (!ValidateResponse(response))
+		return;
+	bool streamVisible = !!response[1].value_union.ui32;
+
+	sid->streamVisibleChanged = false;
```

Comment:
Best to verify if `sid` is valid before changing its values

Reply:
Good catch! Fixed 👍